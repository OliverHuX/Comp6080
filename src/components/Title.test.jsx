import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Title from './Title';

describe('Title Component', () => {
  it('has props children', () => {
    const title = shallow(<Title>title string</Title>);
    title.contains('title string').valueOf(true);
  });
  it('has no rops children', () => {
    const title = shallow(<Title></Title>);
    title.contains('title string').valueOf(false);
  });
  it('renders with minimal props (snapshot)', () => {
    const title = renderer.create(<Title></Title>).toJSON();
    expect(title).toMatchSnapshot();
  });
  it('renders with props (snapshot)', () => {
    const title = renderer.create(<Title>Sample title</Title>).toJSON();
    expect(title).toMatchSnapshot();
  });
})
