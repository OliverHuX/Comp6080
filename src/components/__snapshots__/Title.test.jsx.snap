// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component renders with minimal props (snapshot) 1`] = `
<h2
  className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
/>
`;

exports[`Title Component renders with props (snapshot) 1`] = `
<h2
  className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
>
  Sample title
</h2>
`;
